src/librustc_typeck/check/method/probe.rs

```diff
@@ -1181,6 +1184,29 @@ impl<'a, 'gcx, 'tcx> ProbeContext<'a, 'gcx, 'tcx> {
         })
     }
 
+    /// Check whether there are methods that are commonly confused. An example
+    /// could be using `Vec::length` instead of `Vec::len`.
+    ///
+    /// In contrast to `probe_for_lev_candidate`, this is based on a manually
+    /// curated whitelist.
+    fn probe_for_whitelist_candidate(&mut self) -> Result<Option<ty::AssociatedItem>, MethodError<'tcx>> {
+        debug!("Probing for method names that are often confused with {:?}",
+               self.method_name);
+
+        let method_name: ast::Name = match self.method_name {
+            Some(name) => name,
+            None => return Ok(None),
+        };
+
+        println!("Method name: {:?}", method_name);
```

Comment:
Please remove these lines.

Reply:
Sure, this is not done at all, the debug prints will be gone when I'm done. I just wanted to collect some initial feedback before spending too much time on it :)

Reply:
No problem. You have a merge conflict by the way. Please rebase

Reply:
I'm aware, but as long as the questions about my general approach are still open (and the tests currently fail anyways) there's not much use in keeping CI busy for multiple hours 🙂

Reply:
btw, if you use `debug!` instead of `println!`, then you can just leave the messages in when you commit, which to me is preferable (I hate when I have to debug some code and it has no `debug!` statements at all, personally)